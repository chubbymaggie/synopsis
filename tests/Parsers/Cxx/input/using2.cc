// From C++WD'96 7.4.3.2 Example
namespace A
{
  int i;
  namespace B
  {
    namespace C
    {
      int i;
    }
    using namespace A::B::C;
    void f1()
    {
      i = 5; // C::i hides A::i
    }
  }
  namespace D
  {
    using namespace B;
    using namespace C;
    void f2()
    {
      i = 5; // ambiguous, B::C::i or A::i
    }
  }
  void f3()
  {
    i = 5; // uses A::i
  }
}

void f4()
{
  i = 5; // ill-formed, neither i visible
}
