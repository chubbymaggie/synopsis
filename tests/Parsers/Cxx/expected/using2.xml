<?xml version='1.0' encoding='ISO-8859-1'?>
<?xml-stylesheet href='/home/stefan/synopsis/Synopsis/share/Synopsis/dump.css' type='text/css'?>
<ast>
 <declarations>
  <instance accessibility="0" class="Synopsis.AST.Module" file="Parsers/Cxx/input/using2.cc" language="C++" line="0" name="A" type="namespace">
   <comments>
    <instance class="Synopsis.AST.Comment" line="0" suspect="0">
     <file>
      Parsers/Cxx/input/using2.cc
     </file>
     <text>
      // From C++WD'96 7.4.3.2 Example

     </text>
    </instance>
   </comments>
   <declarations>
    <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/using2.cc" language="C++" line="4" name="A.i" type="namespace variable">
     <vtype>
      <instance class="Synopsis.Type.Base" language="C++" name="int"/>
     </vtype>
    </instance>
    <instance accessibility="0" class="Synopsis.AST.Module" file="Parsers/Cxx/input/using2.cc" language="C++" line="0" name="A.B" type="namespace">
     <declarations>
      <instance accessibility="0" class="Synopsis.AST.Module" file="Parsers/Cxx/input/using2.cc" language="C++" line="0" name="A.B.C" type="namespace">
       <declarations>
        <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/using2.cc" language="C++" line="9" name="A.B.C.i" type="namespace variable">
         <vtype/>
        </instance>
        <instance accessibility="0" class="Synopsis.AST.Declaration" file="Parsers/Cxx/input/using2.cc" language="C++" line="9" name="dummy" type="dummy"/>
       </declarations>
      </instance>
      <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/using2.cc" language="C++" line="12" name="A.B.f1()" realname="f1" type="function">
       <returnType>
        <instance class="Synopsis.Type.Base" language="C++" name="void"/>
       </returnType>
      </instance>
      <instance accessibility="0" class="Synopsis.AST.Declaration" file="Parsers/Cxx/input/using2.cc" language="C++" line="12" name="dummy" type="dummy"/>
     </declarations>
    </instance>
    <instance accessibility="0" class="Synopsis.AST.Module" file="Parsers/Cxx/input/using2.cc" language="C++" line="0" name="A.D" type="namespace">
     <declarations>
      <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/using2.cc" language="C++" line="21" name="A.D.f2()" realname="f2" type="function">
       <returnType/>
      </instance>
      <instance accessibility="0" class="Synopsis.AST.Declaration" file="Parsers/Cxx/input/using2.cc" language="C++" line="21" name="dummy" type="dummy"/>
     </declarations>
    </instance>
    <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/using2.cc" language="C++" line="26" name="A.f3()" realname="f3" type="function">
     <returnType/>
    </instance>
    <instance accessibility="0" class="Synopsis.AST.Declaration" file="Parsers/Cxx/input/using2.cc" language="C++" line="26" name="dummy" type="dummy"/>
   </declarations>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/using2.cc" language="C++" line="32" name="f4()" realname="f4" type="function">
   <returnType/>
  </instance>
 </declarations>
 <types>
  <instance class="Synopsis.Type.Declared" language="C++" name="A.D.f2()">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="dummy">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="A">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="f4()">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="A.B.C">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="A.i">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="A.B.C.i">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="A.f3()">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="A.B">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="A.B.f1()">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="A.D">
   <declaration/>
  </instance>
 </types>
 <files>
  <file>
   Parsers/Cxx/input/using2.cc
  </file>
 </files>
</ast>
